docs/blueprint/EP039.rst

```diff
@@ -0,0 +1,280 @@
+        headers=[("source", "kytos".encode())],  # Custom headers
+        timestamp_ms=int(time.time() * 1000)  # Timestamp in ms
+    )
+
+5. How Kafka Stores the Message
+--------------------------------
+Kafka writes the message to a partition inside a topic. The message is stored in **binary format** with an **offset**:
+
+.. code-block:: text
+
+    Topic: kytos_events, Partition: 2
+    --------------------------------------------------
+    Offset | Key   | Value (JSON)                    | Timestamp
+    --------------------------------------------------
+    1023   | None  | { "event": "kytos/..." }        | 1707859200000
+    1024   | None  | { "event": "kytos/..." }        | 1707859210000
+
+6. Message Retrieval (Kafka Consumer Example)
+----------------------------------------------
+A consumer reads the message and decodes it:
+
+.. code-block:: python
+
+    async for msg in consumer:
+        event = json.loads(msg.value.decode())
+        print(f"Received: {event}")
+
+
+XIII. Implementation details ``v1``
+===================================
+
+The following requirements clarify certain details and expected behavior for ``kafka_napp`` v1:
+
+1. Initialization (setup())
+  - Log the startup process: "SETUP Kytos/Kafka"
+  - Create Kafka Producer and Admin Client:
+  - Instantiate KafkaSendOperations, which sets up:
+    - _setup_admin(): Creates a KafkaAdminClient, retrying up to 3 times if Kafka isn't available.
+  - Instantiate AsyncScheduler:
+    - Starts an async loop in a separate thread.
+    - Runs setup_dependencies() in the async loop:
+      - Calls start_up(): Initializes an AIOKafkaProducer with retries.
+      - Checks if the topic exists (check_for_topic()), creating it if necessary (create_topic()).
+      - Run setup_dependencies() asynchronously using run_callable_soon(), so Kytos doesn't block.
+
+2. Event Handling (handle_new_switch())
+  - Triggered when a switch connects (or other event matches the regex pattern .*).
+  - Check if the event is in IGNORED_EVENTS; if so, ignore it.
+  - Send the event data to Kafka via _send_ops.send_message():
+  - Calls run_coroutine(), ensuring the coroutine runs inside the async thread.
+
+  - send_message():
+    - Converts the event to JSON.
+    - Uses AIOKafkaProducer.send() to publish it to the Kafka topic.
+
+3. Shutdown (shutdown())
+  - Log the shutdown process: "SHUTDOWN Kafka/Kytos"
+  - Stop the async event loop:
+  - run_coroutine(self._async_loop.stop())
+  - Wait for the thread to finish using close_thread(), ensuring graceful shutdown.
+  - This implicitly shuts down the Kafka producer (_send_ops.shutdown()) as provided in setup().
+
+XIV. Open Questions / Future Work
+=================================
+
+  1. Error codes
+  2. Specifying what events are ignored
+  3. When sending messages, type and event are equal.
+    - Potential trimming of network I/O by removing one.
+  4. Abstracting away from Kafka to allow any message broker.
```

Comment:
You can nuke 4 here, any other future broker we should consider a new NApp acting as new adapter. Our goal here is indeed to only support what Kafka provides and be as close to broker client (aiokafka) as possible without overabstracting major parts. Keeping it dead simple to maintain too. If we ever need to swap the current client we'll refactor minor parts too since most clients are very similar, and that won't break the NApp functionality, and don't want to pay the price to abstract over it either.